compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -2258,8 +2257,9 @@ private void createObjectTypeTypeDef(BLangTypeDefinition typeDefinition,
             objFieldInfo.fieldType = objField.type;
 
             // Populate default values
-            if (objField.expr != null && (objField.expr.getKind() == NodeKind.LITERAL &&
-                    objField.expr.type.getKind() != TypeKind.ARRAY)) {
+            if (objField.expr != null && (objField.expr.getKind() == NodeKind.LITERAL ||
+                    objField.expr.getKind() == NodeKind.NUMERIC_LITERAL) &&
+                    objField.expr.type.getKind() != TypeKind.ARRAY) {
```

Comment:
shall we move this (condition) to a util method? can re-use it at L#2346 as well.